### DISCUSSION TITLE
Quadlets and image pull secrets

### DISCUSSION THREAD

**Discussion by christian-2** [containers/podman#21242] [Q&A]
Can Podman quadlets pull images that require image pull secrets ? I've seen no suitable option in `quadlet(5)`, however this seems like a frequent and relevant use case. How can it be addressed?

**Comment by ygalblum** [comment]
Thanks for raising this.

IIRC there was already a discussion about this and a similar one about supporting the creation of secrets in general.
The main issue is where would you store the text data. Once we decide on a solution for this, the support could be added. For example, by adding a `.registry` file that other Quadlet files can depend one.

Did you have something in mind?

**Reply by christian-2** [reply]
Thanks for your reply. For my purposes it would be sufficient if systemd `.container` (or for that matter `registry`) units would support an `ImagePullSecret=`. I would then place this option in a systemd drop-in unit with suitably restricted access rights.
